Tests/SwiftAlgorithmsTests/UniqueTests.swift

```diff
@@ -17,10 +17,10 @@ final class UniqueTests: XCTestCase {
     let a = repeatElement(1...10, count: 15).joined().shuffled()
     let b = a.uniqued()
     XCTAssertEqual(b.sorted(), Set(a).sorted())
```

Comment:
Nit: Maybe is worth a test case with all elements equal? e.g. `Array(repeating: 1, count: 10)`, I think this would not test anything different but is good for exercising the worst case scenario of iterators `next()`. Do you think that makes sense?

Reply:
That's reasonable, I've added tests for arrays with a single repeated element 👍